src/main/java/com/fauna/client/FaunaClient.java

```diff
@@ -458,4 +490,58 @@ public <E> FaunaStream<E> stream(Query fql, Class<E> elementClass) {
         return completeAsync(asyncStream(fql, elementClass), STREAM_SUBSCRIPTION);
     }
     //endregion
+
+    //region Event Feeds
+
+    /**
+     * Send a request to the Fauna feed endpoint, and return a CompletableFuture that completes with the response.
+     * @param feedRequest   The Feed Request object.
+     * @param elementClass  The expected class &lt;E&gt; of the feed events.
+     * @return FeedSuccess  A CompletableFuture that completes with the successful feed response.
+     * @param <E>           The type of the feed events.
+     */
+    public <E> CompletableFuture<FeedSuccess<E>> asyncFeed(FeedRequest feedRequest, Class<E> elementClass) {
+        return new RetryHandler<FeedSuccess<E>>(getRetryStrategy(), logger).execute(makeAsyncFeedRequest(
+                getHttpClient(), getFeedRequestBuilder().buildFeedRequest(feedRequest), codecProvider.get(elementClass)));
+    }
+
+    /**
+     * Return a CompletableFuture that completes with a FeedIterator based on an FQL query. This method sends two
+     * requests, one to the query endpoint to get the stream/feed token, and then another request to the feed endpoint
+     * to get the first page of results.
+     * @param fql           The FQL query to be executed. It must return a token, e.g. ends in `.changesOn()`.
+     * @param startTs       Stream start time in microseconds since the Unix epoch.
+     * @param elementClass  The expected class &lt;E&gt; of the feed events.
+     * @return FeedIterator A CompletableFuture that completes with a feed iterator that returns pages of Feed events.
+     * @param <E>           The type of the feed events.
+     */
+    public <E> CompletableFuture<FeedIterator<E>> asyncFeed(Query fql, long startTs, Class<E> elementClass) {
+        return this.asyncQuery(fql, StreamTokenResponse.class).thenApply(str -> this.feed(FeedRequest.builder(str.getData().getToken()).startTs(startTs).build(), elementClass));
+    }
+
+    /**
+     * Return a FeedIterator based on an FQL query. This method sends two requests, one to the query endpoint to get
+     * the stream/feed token, and then another request to the feed endpoint to get the first page of results.
+     * @param fql           The FQL query to be executed. It must return a token, e.g. ends in `.changesOn()`.
+     * @param startTs       Stream start time in microseconds since the Unix epoch.
+     * @param elementClass  The expected class &lt;E&gt; of the feed events.
+     * @return FeedIterator An iterator that returns pages of Feed events.
+     * @param <E>           The type of the feed events.
+     */
+    public <E> FeedIterator<E> feed(Query fql, long startTs, Class<E> elementClass) {
```

Comment:
if i use this API to get an iterator, it's not clear to me how i'd track the cursor and continue my feed where i left off. in python there's a FeedPage that's returned that allows for this. i think with this API it ends up being a pagination until the end of the feed, but then you cannot continue where you left off on the next run (unless i'm missing something?)

Reply:
I guess I was thinking that if you want to continue waiting for new events, that's not a feed, that's a stream. I think you can get the latest TS off the feed events and restart that way but maybe this use case is something to flesh out in the E2E tests.

Reply:
well, yeah, but it's polling instead of streaming. so if i'm running a lambda and i want to poll a feed for the events since the last event i saw, i need a way of persisting data about the last event i saw (whether that's a cursor or a start ts).

Reply:
I think we're actually okay here. I was looking at this and for some reason missing that the iterator returns pages (FeedSuccess) or on flatten() events. Both of these contain the cursor.